### PR TITLE
refactor: migrate & upgrade to webpack v5

### PR DESCRIPTION
**Pull request by bigopon**
## Refactoring

- [x] (1) AureliaDependenciesPlugin
  - [x] upgrade typings
  - [x] upgrade code
- [x] (2) BaseIncludePlugin
  - [x] upgrade typings
  - [x] upgrade code
- [x] (3) ConventionDependenciesPlugin
  - [x] upgrade typings
  - [x] upgrade code
- [x] (4) DistPlugin
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
    - Changes inside this plugin are around how we tap into module resolution & override the path related to dist format
- [x] (5) GlobDependenciesPlugin
  - [x] upgrade typings
  - [x] upgrade code
- [x] (6) html-requires-loader
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
- [x] (7) HtmlDependenciesPlugin
  - [x] upgrade typings
  - [x] upgrade code
- [x] (8) IncludeDependency
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
- [x] (9) InlineViewDependenciesPlugin
  - [x] upgrade typings
  - [x] upgrade code
- [x] (10) ModuleDependenciesPlugin
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
- [x] (11) PreserveExportsPlugin
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
    - from the link referenced by https://github.com/aurelia/webpack-plugin/pull/169#issuecomment-816669758 , it seems either:
      - use `moduleGraph.getIncomingConnections` to get the connections & figured out how to instruct webpack about dependency
- [x] (12) PreserveModuleNamePlugin
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
    - The typing upgrade results in code path that is not clear how to handle.
- [x] (13) SubFolderPlugin
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
    - There are changes related to the hooks used, also the way resolver resolving the modules. Needs more verification around `resolver.doResolve` calls
- [x] (14) **MAIN**: AureliaPlugin
  - [x] upgrade typings
  - [x] upgrade code
  - Comments:
    - webpack 5 no longer exposes modules object and module cache object on the `__webpack_require__` function, add a runtime module to expose them to avoid breaking changes in the `aurelia-loader-webpack`

## 🧪 Test plan
Create different test applications using different configuration for individual plugins, and the Aurelia Plugin, and the test script will invoke webpack to build the applications.

Added test apps:
- Simple application with basic configuration (no routing)
- Simple application with dist es2015 configuration (no routing)
- Simple application with routing & lazy loading + chunk splitting

### CODE REVIEW

**Comment by Expaso** [comment]
Jeeej! Finally it's going to happen! Thank you!

**Comment by bigopon** [comment]
@Expaso still far from the finishing line, but with the latest commit, now it can builds a bundle that works

**Comment by bigopon** [comment]
With fdae3aa, it now can build correctly with normal usages.
Test app configuration can be seen at
- https://github.com/aurelia/webpack-plugin/tree/fdae3aab2238bbf61a6269853540a820eb10f115/tests/app-simple
Webpack configuration for the test app can be seen at
- https://github.com/aurelia/webpack-plugin/blob/fdae3aab2238bbf61a6269853540a820eb10f115/tests/app-simple/webpack.config.js

File: src/PreserveModuleNamePlugin.ts [hunk 1]
```diff
@@ -46,37 +59,43 @@ export class PreserveModuleNamePlugin {
...
-          
+
           id = id.replace(/\\/g, "/");
           if (module.buildMeta)  // meta can be null if the module contains errors
             module.buildMeta["aurelia-id"] = id;
-          if (!this.isDll)
-            module.id = id;
+
+          if (!this.isDll) {
+            compilation.chunkGraph.setModuleId(module, id);
```
Comment (bigopon) [inline src/PreserveModuleNamePlugin.ts:78]: using this should help us being forward compatible with webpack v6, and `module.id` is deprecated

**Review by bigopon — COMMENTED** [review]
Ready for review.

@jods4 can you help review? I've also include dist files to make it convenient for folks to install and try from this branch, please ignore them.

cc @fkleuver @EisenbergEffect @3cp

**Comment by bigopon** [comment]
Things are looking good, i've added & tested another example with a 3rd party plugin that pulls a 3rd party lib: aurelia-chart https://github.com/grofit/aurelia-chart
Merging this.

**Pull request merged**